bin/si/src/main.rs

```diff
@@ -41,14 +47,20 @@ async fn main() -> Result<()> {
         std::path::Path::new("/var/run/docker.sock").to_path_buf(),
     ];
 
-    let docker_socket = docker_socket_candidates
-        .iter()
-        .find(|candidate| candidate.exists())
-        .ok_or(eyre!(
+    let docker: DockerClient;
+    if let "" = docker_sock.as_str() {
+        let socket = docker_socket_candidates
+            .iter()
+            .find(|candidate| candidate.exists())
+            .ok_or(eyre!(
             "failed to determine Docker socket location; candidates={docker_socket_candidates:?}"
```

Comment:
Might make sense to add a note to this error about passing in --si-docker-sock or setting the environment variable

Reply:
```

Backtrace omitted. Run with RUST_BACKTRACE=1 environment variable to display it.
Run with RUST_BACKTRACE=full to include source snippets.
Error:
   0: failed to determine Docker socket location. Set a custom location using `--docker_sock` or `SI_DOCKER_SOCK`; candidates=["/Users/stack72/.docker/run/docket.sock"]

Location:
   <unknown>
   ```